File: backend/src/domain/ai_generation/schema.rs
#![allow(dead_code)]
use serde::{Deserialize, Serialize};

// AI Generation domain schemas

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Generator {
    pub id: i64,
    pub name: String,
    pub provider: String,
    pub model: String,
    pub api_key: Option<String>,
    pub api_endpoint: Option<String>,
    pub enabled: Option<bool>,
    pub created_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AIProvider {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub api_key: Option<String>,
    pub api_endpoint: Option<String>,
    pub enabled: Option<bool>,
    pub created_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AIModel {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub provider: String,
    pub enabled: Option<bool>,
    pub created_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct PromptTemplate {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub template: String,
    pub input_variables: Option<String>,
    pub created_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GenerateTask {
    pub id: i64,
    pub generator_id: i64,
    pub input: String,
    pub output: Option<String>,
    pub status: Option<String>,
    pub error: Option<String>,
    pub created_at: Option<String>,
    pub started_at: Option<String>,
    pub completed_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetGeneratorReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetGeneratorRes {
    pub generator: Generator,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchGeneratorsReq {
    pub provider: Option<String>,
    pub model: Option<String>,
    pub enabled: Option<bool>,
    pub page: i64,
    pub limit: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchGeneratorsRes {
    pub generators: Vec<Generator>,
    pub total: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddGeneratorReq {
    pub name: String,
    pub provider: String,
    pub model: String,
    pub api_key: Option<String>,
    pub api_endpoint: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddGeneratorRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateGeneratorReq {
    pub id: i64,
    pub name: Option<String>,
    pub provider: Option<String>,
    pub model: Option<String>,
    pub api_key: Option<String>,
    pub api_endpoint: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateGeneratorRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GenerateContentReq {
    pub generator_id: i64,
    pub input: String,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GenerateContentRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetPromptTemplateReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetPromptTemplateRes {
    pub prompt_template: PromptTemplate,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchPromptTemplatesReq {
    pub name: Option<String>,
    pub provider: Option<String>,
    pub page: i64,
    pub limit: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchPromptTemplatesRes {
    pub prompt_templates: Vec<PromptTemplate>,
    pub total: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddPromptTemplateReq {
    pub name: String,
    pub description: Option<String>,
    pub template: String,
    pub input_variables: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddPromptTemplateRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdatePromptTemplateReq {
    pub id: i64,
    pub name: Option<String>,
    pub description: Option<String>,
    pub template: Option<String>,
    pub input_variables: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdatePromptTemplateRes {
    pub id: i64,
}

File: backend/domain/ai_generation/schema.go
package ai_generation

import (
	"time"
)

type Generator struct {
	ID          int64      `json:"id"`
	Name        string     `json:"name"`
	Provider    string     `json:"provider"`
	Model       string     `json:"model"`
	APIKey      string     `json:"api_key"`
	APIEndpoint string     `json:"api_endpoint"`
	Enabled     bool       `json:"enabled"`
	CreatedAt   *time.Time `json:"created_at"`
}

type PromptTemplate struct {
	ID             int64      `json:"id"`
	Name           string     `json:"name"`
	Description    string     `json:"description"`
	Template       string     `json:"template"`
	InputVariables string     `json:"input_variables"`
	CreatedAt      *time.Time `json:"created_at"`
}

type GenerateTask struct {
	ID          int64      `json:"id"`
	GeneratorID int64      `json:"generator_id"`
	Input       string     `json:"input"`
	Output      string     `json:"output"`
	Status      string     `json:"status"`
	Error       string     `json:"error"`
	CreatedAt   *time.Time `json:"created_at"`
	StartedAt   *time.Time `json:"started_at"`
	CompletedAt *time.Time `json:"completed_at"`
}


type GetGeneratorReq struct {
	ID int64 `json:"id" binding:"required"`
}

type GetGeneratorRes struct {
	Generator *Generator `json:"generator"`
}

type SearchGeneratorsReq struct {
	Provider string `json:"provider"`
	Model    string `json:"model"`
	Enabled  *bool  `json:"enabled"`
	Page     int64  `json:"page"`
	Limit    int64  `json:"limit"`
}

type SearchGeneratorsRes struct {
	Generators []*Generator `json:"generators"`
	Total      int64        `json:"total"`
}

type AddGeneratorReq struct {
	Name        string `json:"name" binding:"required"`
	Provider    string `json:"provider" binding:"required"`
	Model       string `json:"model" binding:"required"`
	APIKey      string `json:"api_key"`
	APIEndpoint string `json:"api_endpoint"`
	Enabled     *bool  `json:"enabled"`
}

type AddGeneratorRes struct {
	ID int64 `json:"id"`
}

type UpdateGeneratorReq struct {
	ID          int64   `json:"id" binding:"required"`
	Name        *string `json:"name"`
	Provider    *string `json:"provider"`
	Model       *string `json:"model"`
	APIKey      *string `json:"api_key"`
	APIEndpoint *string `json:"api_endpoint"`
	Enabled     *bool   `json:"enabled"`
}

type UpdateGeneratorRes struct {
	ID int64 `json:"id"`
}

type GenerateContentReq struct {
	GeneratorID int64  `json:"generator_id" binding:"required"`
	Input       string `json:"input" binding:"required"`
}

type GenerateContentRes struct {
	ID int64 `json:"id"`
}

type GetPromptTemplateReq struct {
	ID int64 `json:"id" binding:"required"`
}

type GetPromptTemplateRes struct {
	PromptTemplate *PromptTemplate `json:"prompt_template"`
}

type SearchPromptTemplatesReq struct {
	Name     string `json:"name"`
	Provider string `json:"provider"`
	Page     int64  `json:"page"`
	Limit    int64  `json:"limit"`
}

type SearchPromptTemplatesRes struct {
	PromptTemplates []*PromptTemplate `json:"prompt_templates"`
	Total           int64             `json:"total"`
}

type AddPromptTemplateReq struct {
	Name           string `json:"name" binding:"required"`
	Description    string `json:"description"`
	Template       string `json:"template" binding:"required"`
	InputVariables string `json:"input_variables"`
}

type AddPromptTemplateRes struct {
	ID int64 `json:"id"`
}

type UpdatePromptTemplateReq struct {
	ID             int64   `json:"id" binding:"required"`
	Name           *string `json:"name"`
	Description    *string `json:"description"`
	Template       *string `json:"template"`
	InputVariables *string `json:"input_variables"`
}

type UpdatePromptTemplateRes struct {
	ID int64 `json:"id"`
}

// ===================== Generate Task API =====================

type GetGenerateTaskReq struct {
	ID int64 `json:"id" binding:"required"`
}

type GetGenerateTaskRes struct {
	GenerateTask *GenerateTask `json:"generate_task"`
}

